pkg/operator/apiservercontrollerset/apiservercontrollerset.go

```diff
@@ -0,0 +1,179 @@
+		operatorClient,
+		apiregistrationInformers,
+		apiregistrationv1Client.ApiregistrationV1(),
+		kubeInformersForTargetNamesace,
+		kubeClient,
+		apiServerControllerSet.eventRecorder,
+	)
+
+	apiServerControllerSet.finalizerController = nsfinalizercontroller.NewFinalizerController(
+		targetNamespace,
+		kubeInformersForTargetNamesace,
+		kubeClient.CoreV1(),
+		apiServerControllerSet.eventRecorder,
+	)
+
+	return apiServerControllerSet
+}
+
+func (cs *APIServerControllerSet) WithEncryptionControllers(
+	encryptionResources []schema.GroupResource,
+
+	configClient configv1client.Interface,
+	dynamicClientForMigration dynamic.Interface,
+	operatorConfigClient *operatorv1client.Clientset,
+
+	configInformers configinformers.SharedInformerFactory,
+	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces,
+
+	resourceSyncer libgoresourcesyncer.ResourceSyncer,
+) *APIServerControllerSet {
+	nodeProvider := DaemonSetNodeProvider{
+		TargetNamespace:                  cs.targetNamespace,
+		TargetNamespaceDaemonSetInformer: kubeInformersForNamespaces.InformersFor(cs.targetNamespace).Apps().V1().DaemonSets(),
+		NodeInformer:                     kubeInformersForNamespaces.InformersFor("").Core().V1().Nodes(),
+	}
+
+	deployer, err := encryptiondeployer.NewRevisionLabelPodDeployer(
+		"revision",
+		cs.targetNamespace,
+		kubeInformersForNamespaces,
+		resourceSyncer,
+		cs.kubeClient.CoreV1(),
+		cs.kubeClient.CoreV1(),
+		nodeProvider,
+	)
+
+	if err != nil {
+		panic(err)
+	}
+
+	migrator := migrators.NewInProcessMigrator(dynamicClientForMigration, cs.kubeClient.Discovery())
+
+	cs.encryptionControllers, err = encryption.NewControllers(
+		cs.targetNamespace,
+		deployer,
+		migrator,
+		cs.operatorClient, // or should it be apiserverOperatorClient
+		configClient.ConfigV1().APIServers(),
+		configInformers.Config().V1().APIServers(),
+		kubeInformersForNamespaces,
+		cs.kubeClient.CoreV1(),
+		cs.eventRecorder,
+		encryptionResources...,
+	)
+
+	if err != nil {
+		panic(err)
+	}
+
+	cs.revisionController = revisioncontroller.NewRevisionController(
+		cs.targetNamespace,
+		nil,
+		[]revision.RevisionResource{{
+			Name:     "encryption-config",
+			Optional: true,
+		}},
+		kubeInformersForNamespaces.InformersFor(cs.targetNamespace),
+		OpenshiftDeploymentLatestRevisionClient{OperatorClient: cs.operatorClient, TypedClient: operatorConfigClient.OperatorV1()},
+		v1helpers.CachedConfigMapGetter(cs.kubeClient.CoreV1(), kubeInformersForNamespaces),
+		v1helpers.CachedSecretGetter(cs.kubeClient.CoreV1(), kubeInformersForNamespaces),
+		cs.eventRecorder,
+	)
+
+	cs.pruneController = prune.NewPruneController(
+		cs.targetNamespace,
+		[]string{"encryption-config-"},
+		cs.kubeClient.CoreV1(),
+		cs.kubeClient.CoreV1(),
+		kubeInformersForNamespaces,
+		cs.eventRecorder,
+	)
+
+	return cs
+}
+
+func (cs *APIServerControllerSet) Run(ctx context.Context) {
+	go cs.apiServiceController.Run(1, ctx.Done())
+	go cs.finalizerController.Run(1, ctx.Done())
```

Comment:
missing loglevel, status, configupgradeable.

Reply:
I can see what we're doing now and how far we're going with the modularity, I'll add those.